tools/project_manager/page_compare.php

```diff
@@ -0,0 +1,182 @@
+<?php
+$relPath="./../../pinc/";
+include_once($relPath.'base.inc');
+include_once($relPath.'misc.inc');
+include_once($relPath.'theme.inc');
+include_once($relPath.'Project.inc');
+include_once($relPath."unformat.inc"); // remove_formatting(), unspace()
+
+require_login();
+
+$projectid = validate_projectID('project', @$_GET['project']);
+
+$project = new Project($projectid );
+
+$state = $project->state;
+$title = $project->nameofwork;
+$sub_title = _('Compare pages with formatting removed');
+
+output_header( "$sub_title: $title", NO_STATSBAR);
+
+echo "<h1>$title</h1>\n";
+echo "<h2>$sub_title</h2>\n";
+
+$project_url = "$code_url/project.php?id=$projectid&amp;expected_state=$state";
+$label = _("Return to Project Page");
+
+echo "<p><a href='$project_url'>$label</a></p>\n";
+
+if (!$project->pages_table_exists)
+{
+    echo "<p>";
+    if ($project->archived != 0) {
+        echo _("The project has been archived, so page details are not available.");
+    } elseif ($project->state == PROJ_DELETE) {
+        echo _("The project has been deleted, so page details are not available.");
+    } else {
+        echo _("Page details are not available for this project.");
+    }
+    echo "</p>";
+    exit();
+}
+
+$L_round_id = array_get($_GET, "L_round_id", "P3");
```

Comment:
Move all $_GET accesses to the top of the file rather than have them scattered throughout. This allows one place to do user-input validation.